jboss/container/maven/default/bash/artifacts/opt/jboss/container/maven/default/maven-repos.sh

```diff
@@ -83,19 +122,18 @@ function configure_maven_repo() {
 }
 
 function configure_maven_repos() {
-    local settings="${1-$HOME/.m2/settings.xml}"
-    local local_repo_path="${MAVEN_LOCAL_REPO}"
-    if [ "${local_repo_path}" != "" ]; then
-        set_local_repo_path "${settings}" "${local_repo_path}"
-    fi
-    # single repo scenario: respect fully qualified url if specified, otherwise find and use service
+    local settings="${1:-${HOME}/.m2/settings.xml}"
+    local local_repo_path="${MAVEN_LOCAL_REPO:-${MAVEN_REPO_LOCAL:-${HOME}/.m2/repository}}"
```

Comment:
`MAVEN_LOCAL_REPO` _is_ the variable that defines the local repository.  It should not be initialized here, but where the env is initialized for Maven.  This change will break other images.  You'll need to initialize the maven environment separately, before calling any of the other functions.  If you need to specialize the environment, look at one of the product specific s2i modules for an example.

Reply:
FWIW, maven_repo_local was deprecated because it overlaps with the prefix for defining maven repos (not that you'd likely have a repo named local, but..) and it reads better.

Reply:
Here's an example of initializing the core maven module:

https://github.com/jboss-openshift/cct_module/blob/master/jboss/container/maven/s2i/bash/artifacts/opt/jboss/container/maven/s2i/maven-s2i#L39-L53

Ignore the maven_s2i_* methods, as those are specific to the maven s2i module.

Here's where it's customizing what goes on in maven_init:

https://github.com/jboss-openshift/cct_module/blob/master/jboss/container/maven/s2i/bash/artifacts/opt/jboss/container/maven/s2i/maven-overrides

HTH

Reply:
Hi @rcernich . I've made changes. Note that now instead of maven.sh depending upon maven-repos.sh, it's the other way around (and maven-repos.sh was renamed to maven-settings.sh). Please see this PR in concert with https://github.com/jboss-container-images/jboss-kie-modules/pull/110 . Thanks!